quickwit/quickwit-ingest-api/src/ingest_api_service.rs

```diff
@@ -96,7 +93,9 @@ impl IngestApiService {
             // TODO better error handling.
             // If there is an error, we probably want a transactional behavior.
             let records_it = iter_doc_payloads(doc_batch);
-            self.queues.append_batch(&doc_batch.index_id, records_it)?;
+            self.queues
```

Comment:
This implementation is properly asynchronous.
We can now have that actor run on the regular runtime, and use the `ctx.protect_future` mechanism to guard us from waiting for the runtime, or io wait.

Reply:
@trinity you need protect_future on this